src/smt/smt_solver.cpp

```diff
@@ -150,6 +151,17 @@ Result SmtSolver::checkSatInternal()
     }
     Trace("smt") << "SmtSolver::global negate returned " << result << std::endl;
   }
+
+  // Handle emitting pending partitions.
```

Comment:
I think this block of code should be moved to `TheoryEngine::postsolve()` which is called from https://github.com/cvc5/cvc5/blob/main/src/prop/prop_engine.cpp#L457.

You'll have to extend this method to pass the result.

Reply:
This is done now and still working as expected.